tests/src/test/java/org/eclipse/hono/tests/amqp/AmqpAdapterBase.java

```diff
@@ -220,13 +255,14 @@ private void setupProtocolAdapter(final String tenantId, final String deviceId,
                     return connectToAdapter(username, DEVICE_PASSWORD, false)
                     .compose(connectedClient -> createProducer(tenantId))
                     .map(s -> {
-                        sender = s;
+                        result.complete(s);
                         return s;
                     });
 
                 }).setHandler(ctx.asyncAssertSuccess(ok -> setup.complete()));
 
         setup.await(2000);
+        return result;
```

Comment:
It doesn't make much sense to wait for an Async now anymore, does it? Instead, the invoking code should simply compose the returned `Future<MessageSender>` with the code that uses the sender ...

Reply:
:+1: